launch: # 起動パラメータ
  ros__parameters:
    joy: false   # ジョイスティックの使用を有効にする
    urg: false   # URG LiDARの使用を有効にする
    rviz: false  # RVizの使用を有効にする
    usb_cam: true

icart_mini_driver_node:
    ros__parameters:
            odom_frame_id: 'odom'   # odomフレームのID
            base_frame_id: 'base_footprint'  # ベースフレームのID
            Hz: 40  # ノードの更新頻度（Hz単位）
            left_wheel_joint: 'left_wheel_joint'    # 左ホイールのジョイント名
            right_wheel_joint: 'right_wheel_joint'   # 右ホイールのジョイント名
            liner_vel_lim: 1.5  # 並進速度の制限（m/s）
            liner_accel_lim: 1.5    # 並進加速度の制限（m/s^2）
            angular_vel_lim: 3.14   # 角速度の制限（rad/s）
            angular_accel_lim: 3.14 # 角加速度の制限（rad/s^2）
            calculate_odom_from_ypspur: true    # YPSPURからのオドメトリ計算を有効にする
            publish_odom_tf: true  # オドメトリのTFをパブリッシュするかどうか


tele_controller_node:
  ros__parameters:
    linear_max_vel: 0.5         # 最大並進速度（m/s）
    angular_max_vel: 1.57       # 最大角速度（rad/s）
    autonomous_flag: false      # 自律走行フラグの初期値

joy_node:
    ros__parameters:
        device: /dev/input/js0  # ジョイスティックデバイスのパス
        deadzone: 0.3  # デッドゾーンの設定（0.0から1.0）
        autorepeat_rate: 20.0  # 自動リピートレート（Hz単位）

livox_lidar_publisher:
  ros__parameters:
    xfer_format: 0              # 0: PointCloud2形式で出力（Rviz2対応）、1: Livox独自形式（カスタムMsg）
    multi_topic: 0              # 0: 全てのLiDARが同じトピックを使用、1: LiDARごとに個別のトピックを使用
    data_src: 0                 # 0: LiDARからの生データ、他は無効（未使用）
    publish_freq: 10.0          # データの発行頻度（Hz）。例: 5.0, 10.0, 20.0など
    output_data_type: 0         # 出力先タイプ（通常は0でROSに出力）
    frame_id: "livox_frame"     # 出力データに使用するTFのフレームID
    lvx_file_path: "/home/livox/livox_test.lvx"  # 録画ファイルのパス（使用しない場合は空でも可）
    # user_config_path: "/your/path/to/MID360_config.json"  # LiDARのネットワークやIP設定などの詳細構成（launchで有効化済み）
    cmdline_input_bd_code: "livox0000000001"     # LiDARデバイスのIDを手動で指定（必要に応じて）

urg_node:
  ros__parameters:
    serial_port: '/dev/sensors/hokuyo'  # シリアルポートの指定
    serial_baud: 115200 # ボーレートの指定
    frame_id : 'laser'  # 出力データのフレームID
    calibrate_time : false  # 時間のキャリブレーションを行うかどうか
    synchronize_time : false    # 時間の同期を行うかどうか
    publish_intensity : false   # 強度データをパブリッシュするかどうか
    publish_multiecho : false   # マルチエコーをパブリッシュするかどうか
    error_limit : 4 # エラーの制限値
    error_reset_period : 5.0    # エラーリセットの周期
    diagnostics_tolerance : 0.05    # 診断の許容値
    diagnostics_window_time : 5.0   # 診断のウィンドウ時間
    time_offset : 0.0   # 時間オフセット
    angle_min : -3.14   # 最小角度
    angle_max : 3.14    # 最大角度
    skip : 0    # スキップするデータ数
    cluster : 1   # クラスタリングの設定


usb_cam_node:
  ros__parameters:
    video_device: "/dev/video4"  # USBカメラのデバイス名
    framerate: 30.0              # フレームレート
    io_method: "mmap"           # 入出力メソッド
    frame_id: "camera"          # フレームID
    pixel_format: "mjpeg2rgb"   # ピクセルフォーマット
    av_device_format: "YUV422P" # デバイスフォーマット
    image_width: 640            # 画像の幅
    image_height: 480           # 画像の高さ
    camera_name: "test_camera"  # カメラ名
    camera_info_url: "package://usb_cam/config/camera_info.yaml"  # カメラ情報のURL
    brightness: -1              # 明るさ
    contrast: -1                # コントラスト
    saturation: -1              # 彩度
    sharpness: -1               # シャープネス
    gain: -1                    # ゲイン
    auto_white_balance: true     # 自動ホワイトバランスを有効にするかどうか
    white_balance: 4000         # ホワイトバランスの値
    autoexposure: true          # 自動露出を有効にするかどうか
    exposure: 100               # 露出の値
    autofocus: false            # オートフォーカスを有効にするかどうか
    focus: -1                   # フォーカスの値